ceph/ceph_admin/apply_spec.py

```diff
@@ -0,0 +1,74 @@
+"""
+Represent the ceph orch CLI action 'apply' using spec file.
+
+Module to deploy ceph service(s) using orchestration command
+"ceph orch apply -i spec.yaml"
+
+"""
+import logging
+import tempfile
+from typing import Dict
+
+from .common import config_dict_to_string
+from .helper import PrepareSpec
+from .typing_ import ServiceProtocol
+
+LOG = logging.getLogger()
+
+
+class ApplySpec:
+    """Add apply command support to provision service using spec file."""
+
+    def apply_spec(self: ServiceProtocol, config: Dict) -> None:
+        """
+        Execute the apply_spec method using the object's service name and provided input.
+
+        Args:
+            config:     Key/value pairs passed from the test suite.
+                        base_cmd_args   - key/value pairs to set for base command
+                        specs           - service specifications.
+
+        Example:
+            config:
+                command: apply_spec
+                service: orch
+                base_cmd_args:          # arguments to ceph orch
+                    concise: true
+                    verbose: true
+                specs:
+                 - service_type: host
```

Comment:
This can be inserted at the individual service.

Reply:
As we discussed, apply_spec is part Orch. Service object is not involved here to support multi doc yaml content.